pxtlib/melody-editor/melodyGallery.ts

```diff
@@ -148,32 +155,77 @@ namespace pxtmelody {
 
             const leftButton = mkElement("div", {
                 className: "melody-editor-button left-button",
-                role: "button",
-                title: sample.name
+                title: sample.name,
+                role: "menuitem",
+                tabIndex: 0,
             }, () => this.handleSelection(sample))
 
             leftButton.appendChild(icon);
             leftButton.appendChild(label);
             leftButton.appendChild(preview);
 
-
             outer.appendChild(leftButton);
 
+            this.selectionButtons[i] = leftButton;
+ 
             const rightButton = mkElement("div", {
                 className: "melody-editor-button right-button",
                 role: "button",
-                title: lf("Preview {0}", sample.name)
+                title: lf("Preview {0}", sample.name),
+                tabIndex: 0,
             }, () => this.togglePlay(sample, i));
 
             const playIcon = mkElement("i", {
                 className: "play icon"
             });
 
-            this.buttons[i] = playIcon;
+            this.playIcons[i] = playIcon;
 
             rightButton.appendChild(playIcon);
             outer.appendChild(rightButton);
 
+            this.previewButtons[i] = rightButton;
+              
+            leftButton.addEventListener("keydown", (e) => {
```

Comment:
We never remove these event listeners, but it seems to be a common pattern in field_melodySandbox.ts...